SharpPcap/DeviceConfiguration.cs

```diff
@@ -44,6 +44,23 @@ public class DeviceConfiguration
 
         public bool? Immediate { get; set; }
 
+        #region File IO
+        /// <summary>
+        /// Reading and writing capture files
+        /// </summary>
+        public string Filename { get; set; }
```

Comment:
This part I don't agree with, since it creates inconsistency in the API, 
* Live devices takes the device name from `PcapInterface` in the constructor.
* File device takes the file name in the device configuration

Reply:
Thoughts on how to improve?

Reply:
* Keep the file specific stuff like filename and file mode in the constructor for now, and keep only shared stuff in the `DeviceConfiguration` (so only add link layer in this PR).
* Make calling `Open` required for file device so the PcapHandle is created during Open, and not in the constructor

This way we keep the API consistency as far as this PR is concerned, and any future PRs will address live and file devices alike.

Reply:
Oooh now I see what you mean on the constructor part. Yeah let me do that and repush.

Reply:
@kayoub5 who else might use the LinkLayerType other than the capture file writer?

Reply:
Some devices support more than one link layer (for example a WiFi adapter could use Ethernet or radio link layers), I have not seen a device myself, but the libpcap API allows this.